core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexTemplateService.java

```diff
@@ -163,10 +154,62 @@ public void onFailure(String source, Throwable t) {
             }
 
             @Override
-            public ClusterState execute(ClusterState currentState) {
+            public ClusterState execute(ClusterState currentState) throws Exception {
                 if (request.create && currentState.metaData().templates().containsKey(request.name)) {
                     throw new IndexTemplateAlreadyExistsException(request.name);
                 }
+
+                boolean indexCreated = false;
+                String dummyIndexPrefix = ".dummy_";
+                try {
+
+                    //create index service for parsing and validating mappings
+                    Settings.Builder settingsBuider = Settings.settingsBuilder()
+                        .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
+                        .put(request.settings);
+
+                    if (request.settings.get(IndexMetaData.SETTING_NUMBER_OF_SHARDS) == null) {
+                        settingsBuider.put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1);
```

Comment:
Ah, my fault, we already check the settings.
I will change to override shards to 1 and replicas to 0 always.

https://github.com/elastic/elasticsearch/blob/c50b22f95f48119213156bdb634925e1be91b794/core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexTemplateService.java#L215